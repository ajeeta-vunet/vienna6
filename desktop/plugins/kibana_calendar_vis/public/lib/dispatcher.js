/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import d3 from 'd3';

export class Dispatcher {
  constructor() {
    this.container = null;
    this.src = null;
    this.dataTarget = null;
    this.config = null;
    this.API = null;
  }

  newMapping() {
    this.src = null;
    this.dataTarget = null;
    return this;
  }

  addAPI(API) {
    this.API = API;
    return this;
  }

  addConfig(config) {
    this.config = config;
    return this;
  }

  addContainer(container) {
    if (container instanceof HTMLElement) {
      this.container = container;
    } else {
      throw new TypeError(`invalid type ${container.toString()}, should be HTMLElement`);
    }

    return this;
  }

  addSource(src) {
    this.src = src;
    return this;
  }

  addDataTarget(dataTarget) {
    if (!/^\[data-.*\]$/.test(dataTarget)) {
      throw new TypeError(`invalid type ${typeof dataTarget}, should be in a form [data-...]`);
    }
    this.dataTarget = dataTarget;
    return this;
  }

  addEvent(event, callback) {
    if (this.source === null || this.dataTarget === null) {
      throw new TypeError(`source or dataTarget is not defined`);
    }

    const selection = d3.selectAll(this.dataTarget);
    const self = this;
    const nodes = this.src !== null ? [].slice.call(this.container.querySelectorAll(this.src)) : [];
    selection.each(function () {
      const element = d3.select(this);
      if (nodes.length !== 0 && !nodes.includes(this)) {
        return;
      }

      if (typeof callback === 'function') {
        const options = {
          API: self.API,
          config: self.config,
          container: self.container,
          source: self.src,
          dataTarget: self.dataTarget
        };
        return element.on(event, callback.bind(this, options));
      }
    });

    return this;
  }

  removeEvent(event) {
    if (this.source === null || this.dataTarget === null) {
      throw new TypeError(`source or dataTarget is not defined`);
    }

    const selection = d3.selectAll(this.dataTarget);
    selection.each(function () {
      const element = d3.select(this);
      element.on(event, null);
    });

    return this;
  }

}
