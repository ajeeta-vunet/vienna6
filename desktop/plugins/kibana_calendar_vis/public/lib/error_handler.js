/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import MarkdownIt from 'markdown-it';
import d3 from 'd3';

const markdownIt = new MarkdownIt({
  html: true,
  linkify: true
});

export const errorContainer = 'visualize-error';

export class CalendarErrorHandler {
  constructor() {
    this.el = undefined;
    this.id = undefined;
  }

  bindEl(el, id) {
    this.el = el;
    this.id = id;
    return this;
  }

  removeError() {
    d3.select(this.el)
      .attr('id', this.id)
      .selectAll(`.${errorContainer}`)
      .remove();
  }

  error(msg) {
    this.removeError();
    const div = d3.select(this.el)
      .append('div')
      .attr('id', this.id)
      // class name needs `chart` in it for the polling checkSize function
      // to continuously call render on resize
      .attr('class', `${errorContainer} chart error`);

    if (msg === 'No results found') {
      div.append('div')
        .attr('class', `text-center ${errorContainer} visualize-chart`)
        .append('div').attr('class', 'item top')
        .append('div').attr('class', 'item')
        .append('h3').html('No data to show');

      div.append('div').attr('class', 'item bottom');
    } else {
      div.append('h4').html(markdownIt.renderInline(msg));
    }
    return div;
  }
}
