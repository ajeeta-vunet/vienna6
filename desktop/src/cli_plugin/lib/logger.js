export default function Logger(settings) {
  const self = this;

  self.previousLineEnded = true;
  self.silent = !!settings.silent;
  self.quiet = !!settings.quiet;
}

Logger.prototype.log = function (data, sameLine) {
  const self = this;

  if (self.silent || self.quiet) return;

  if (!sameLine && !self.previousLineEnded) {
    process.stdout.write('\n');
  }

  //if data is a stream, pipe it.
  if (data.readable) {
    data.pipe(process.stdout);
    return;
  }

  process.stdout.write(data);
  if (!sameLine) process.stdout.write('\n');
  self.previousLineEnded = !sameLine;
};

Logger.prototype.error = function (data) {
  const self = this;

  if (self.silent) return;

  if (!self.previousLineEnded) {
    process.stderr.write('\n');
  }

  //if data is a stream, pipe it.
  if (data.readable) {
    data.pipe(process.stderr);
    return;
  }
  process.stderr.write(`${data}\n`);
  self.previousLineEnded = true;
};
