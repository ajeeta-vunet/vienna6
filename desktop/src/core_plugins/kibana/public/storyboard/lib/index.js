export { saveStoryboard } from './save_storyboard';
//export { getAppStateDefaults } from './get_app_state_defaults';
