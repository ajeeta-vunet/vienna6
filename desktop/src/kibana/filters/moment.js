define(function (require) {
	var moment = require('moment');
  var _ = require('lodash');

  require('modules')
    .get('kibana/filters')
    .filter('moment', function () {
    return function (datetime, roundUp) {
      var format = 'MMMM Do YYYY, HH:mm:ss.SSS';
      if (moment.isMoment(datetime)) return datetime.format(format);
      if (_.isDate(datetime)) return moment(datetime).format(format);
      return datetime;
    };
  });
});