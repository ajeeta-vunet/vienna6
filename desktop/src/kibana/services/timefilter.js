define(function (require) {
  var _ = require('lodash');
  var moment = require('moment');
  var datemath = require('utils/datemath');
  var module = require('modules').get('kibana');
  require('components/state_management/global_state');

  module.service('timefilter', function (Promise, globalState, $rootScope) {

    var self = this;

    // TODO: This should be disabled on route change, apps need to enable it explicitly
    var enable = false;

    // These can be date math strings or moments.
    $rootScope.time = this.time = _.defaults(globalState.time || {}, {
      from: 'now-15m',
      to: 'now'
    });

    // Only gets called if the user manually updates the URL
    globalState.onUpdate(function readFromGlobalState() {
      _.assign(self.time, globalState.time);
      castTime();
    });

    var init = function () {
      castTime();
    };

    var convertISO8601 = function (stringTime) {
      var obj = moment(stringTime, 'YYYY-MM-DDTHH:mm:ss.SSSZ', true);
      return obj.isValid() ? obj : stringTime;
    };

    var castTime = function () {
      var time = globalState.time;
      if (time && time.from) self.time.from = convertISO8601(time.from);
      if (time && time.to) self.time.to = convertISO8601(time.to);
    };

    this.enabled = function (state) {
      if (!_.isUndefined(state)) enable = state;
      return enable;
    };

    this.get = function (indexPattern) {
      var filter;
      var timefield = indexPattern.timeFieldName && _.find(indexPattern.fields, {name: indexPattern.timeFieldName});

      if (timefield) {
        var bounds = this.getBounds();
        filter = {range : {}};
        filter.range[timefield.name] = {
          gte: bounds.min,
          lte: bounds.max
        };
      }
      return filter;
    };

    this.getBounds = function (timefield) {
      return {
        min: datemath.parse(self.time.from),
        max: datemath.parse(self.time.to, true)
      };
    };

    init();

  });

});
