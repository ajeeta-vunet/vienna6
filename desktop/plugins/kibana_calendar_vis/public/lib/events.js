/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import $ from 'jquery';
import moment from 'moment';

export function highlight({ config, container, dataTarget }) {
  const label = this.getAttribute(dataTarget.replace(/\=".*"/, '').slice(1, -1));

  function justifyOpacity(opacity) {
    const decimalNumber = parseFloat(opacity, 10);
    const fallbackOpacity = 0.5;
    return (0 <= decimalNumber && decimalNumber <= 1) ? decimalNumber : fallbackOpacity;
  }

  const dimming = config.get('visualization:dimmingOpacity');
  $(container).find(dataTarget)
    .css('opacity', 1)
    .not((els, el) => el.getAttribute(dataTarget.slice(1, -1)) === label)
    .css('opacity', justifyOpacity(dimming));
}

export function unHighlight({ container, dataTarget }) {
  const div = $(container);
  $(dataTarget, div).css('opacity', '');
}

export function expandView({ API }) {
  const { timeFilter } = API;
  const { year, month, day } = this.dataset;
  let start;
  let end;
  if (day === undefined) {
    start = moment(new Date(year, month - 1, 1));
    end = moment(new Date(new Date(year, month, 1).getTime() - 1));
  } else {
    start = moment(new Date(year, month - 1, day, 0));
    end = moment(new Date(new Date(year, month - 1, day, 24).getTime() - 1));
  }

  timeFilter.setTime(start, end, 'absolute');
}
