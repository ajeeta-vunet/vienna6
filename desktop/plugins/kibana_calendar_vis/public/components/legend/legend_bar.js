/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import _ from 'lodash';
import React from 'react';
import { Legend } from './legend';
import { legendPosition } from './index';
import { highlight, unHighlight } from '../../lib/events';

export class LegendBar extends React.Component {

  constructor(props) {
    super(props);
    this.legend = undefined;
    this.state = {
      enableHover: this.props.visConfig.get('enableHover'),
      labels: [{
        label: 'loading ...'
      }]
    };
    const { dispatcher } = this.props;
    dispatcher.newMapping().addDataTarget('[data-label]');
  }

  _accommodateContainer() {
    let container;
    if(this.legend) {
      container = this.legend.parentNode.parentNode;
      const legendPos = this.props.position;
      for (const className of container.classList) {
        if (Object.values(legendPosition).includes(className)) {
          container.classList.remove(className);
          break;
        }
      }
      container.classList.add(legendPos);
    }
  }

  highlight() {
    const { dispatcher } = this.props;
    if (this.state.enableHover) {
      dispatcher.addEvent('mouseenter', highlight);
    } else {
      dispatcher.removeEvent('mouseenter');
    }
  }

  unHighlight() {
    const { dispatcher } = this.props;
    if (this.state.enableHover) {
      dispatcher.addEvent('mouseleave', unHighlight);
    } else {
      dispatcher.removeEvent('mouseleave');
    }
  }

  componentWillMount() {
    const { visConfig, getUiState } = this.props;
    const labels = visConfig.get('legend.labels', null);
    const legendOpen = getUiState('vis.legendOpen');
    if (!labels) {
      this.setState({
        labels: [{
          label: 'loading ...'
        }]
      });
    }
    else {
      this.setState({
        open: legendOpen,
        enableHover: visConfig.get('enableHover'),
        labels: _.map(labels, label => ({ label: label }))
      });
    }
  }

  componentDidMount() {
    this._accommodateContainer();
    this.props.renderComplete();
  }

  componentDidUpdate() {
    this._accommodateContainer();
    this.props.renderComplete();
  }

  render() {
    return (
      <div ref={e => {this.legend = e;}} className="legend-col-wrapper">
        <ul className="legend-ul">
          {this.state.labels.map((legendData, i) => (
            <li
              className="legend-value color"
              key={i}
              onMouseEnter={this.highlight.bind(this)}
              onMouseLeave={this.unHighlight.bind(this)}
              data-label={legendData.label}
            >
              <Legend
                legendData={legendData}
                colorFunc={this.props.colorFunc}
              />
            </li>
          ))}
        </ul>
      </div>
    );
  }

  componentWillUnmount() {
    this.legend = null;
  }
}
