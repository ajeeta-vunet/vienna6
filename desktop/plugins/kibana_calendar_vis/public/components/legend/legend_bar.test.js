/* eslint-disable */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import moment from 'moment';
import { mount } from 'enzyme';
import sinon from 'sinon';
import { LegendBar } from './legend_bar';
import { legendPosition } from './';
import { CalendarVisConfig, CalendarErrorHandler } from '../../lib';
import { calendarDataObjectProvider } from '../../lib/data_object/calendar_data';
import { Dispatcher } from '../../lib/dispatcher';
import aggResponse from '../../__tests__/agg_response.json';
import colorMap from '../../__tests__/colormap.json';
import { containerName, legendName, defaultParams } from '../../default_settings';
import { calculateBounds } from 'ui/timefilter/get_time';

describe('LegendBar', () => {

  let visConfig;
  let DataObject;
  let vislibData;
  let dispatcher;
  let visData;
  let legendBarWrapper;
  let renderComplete;
  const fakeConfig = {
    get(key) {
      if (key === 'visualization:colorMapping') {
        return {
          Count: '#00A69B'
        };
      } else if (key === 'visualization:dimmingOpacity') {
        return 0.5;
      }
    }
  };
  const fakeVis = {
    params: defaultParams,
    API: {
      timeFilter: {
        getBounds: function () {
          return calculateBounds({
            from: moment(1531026000000),
            to: moment(1531026000000 + 35 * 86400000)
          });
        }
      }
    }
  };
  let fakeUiState;
  let setUiState;
  let getUiState;

  beforeEach(() => {
    visData = aggResponse;
    visConfig = new CalendarVisConfig(fakeVis, defaultParams);

    visConfig.set('legend', {
      labels: Object.keys(colorMap)
    });

    fakeUiState = {
      get(state) {
        return this[state];
      },
      set(state, value) {
        this[state] = value;
      }
    };
    setUiState = function (state, value) {
      fakeUiState.set(state, value);
    };
    getUiState = function (state) {
      return fakeUiState.get(state);
    };
    fakeUiState.set('vis.legendOpen', true);
    fakeUiState.set('vis.defaultColors', colorMap);

    DataObject = calendarDataObjectProvider(fakeConfig);
    vislibData = new DataObject(visData, fakeUiState);
    dispatcher = new Dispatcher(CalendarErrorHandler.bindEl(document.createElement('div')));
    dispatcher.addConfig(fakeConfig);
    renderComplete = jest.fn();
    legendBarWrapper = mount(
      <div className={containerName}>
        <div className={legendName}>
          <LegendBar
            visConfig={visConfig}
            colorFunc={vislibData.getColorFunc()}
            position={legendPosition[defaultParams.legendPosition]}
            dispatcher={dispatcher}
            setUiState={setUiState}
            getUiState={getUiState}
            renderComplete={renderComplete}
          />
        </div>
      </div>
    );
  });

  afterEach(() => {
    visData = null;
    visConfig = null;
    vislibData = null;
    legendBarWrapper.unmount();
  });

  it('should render a full LegendBar component, default position to the right', () => {
    sinon.spy(LegendBar.prototype, 'componentDidMount');
    expect(renderComplete.mock.calls.length).toBe(1);
    expect(legendBarWrapper.find('li.legend-value')).toHaveLength(4);
    expect(legendBarWrapper).toMatchSnapshot();
  });

  it('should collapse and expand when click the arrow', () => {
    legendBarWrapper.find('button').simulate('click');
    expect(legendBarWrapper.find(LegendBar).instance().state.open).toEqual(false);
    expect(legendBarWrapper).toMatchSnapshot();

    legendBarWrapper.find('button').simulate('click');
    expect(legendBarWrapper.find(LegendBar).instance().state.open).toEqual(true);
  });

  it('should dim and undim the data labels when mouseover and mouseout the legends', () => {
    visConfig.set('enableHover', true);
    sinon.spy(dispatcher, 'addEvent');
    legendBarWrapper = mount(
      <div className={containerName}>
        <div className={legendName}>
          <LegendBar
            visConfig={visConfig}
            colorFunc={vislibData.getColorFunc()}
            position={legendPosition[defaultParams.legendPosition]}
            dispatcher={dispatcher}
            setUiState={setUiState}
            getUiState={getUiState}
            renderComplete={renderComplete}
          />
        </div>
      </div>
    );

    legendBarWrapper.find('li.legend-value').at(0).simulate('mouseenter');
    expect(dispatcher.addEvent.calledOnce).toEqual(true);

    legendBarWrapper.find('li.legend-value').at(0).simulate('mouseleave');
    expect(dispatcher.addEvent.calledTwice).toEqual(true);
  });

});
