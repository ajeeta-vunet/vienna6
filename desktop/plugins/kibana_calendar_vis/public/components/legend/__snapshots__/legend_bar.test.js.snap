// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendBar should collapse and expand when click the arrow 1`] = `
<div
  className="vislib-container calendar-vis-container"
>
  <div
    className="visualize-legend calendar-vis-legend"
  >
    <LegendBar
      colorFunc={[Function]}
      dispatcher={
        Dispatcher {
          "API": null,
          "config": Object {
            "get": [Function],
          },
          "container": null,
          "dataTarget": "[data-label]",
          "src": null,
        }
      }
      getUiState={[Function]}
      position="vis-container--legend-right"
      renderComplete={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
        }
      }
      setUiState={[Function]}
      visConfig={
        CalendarVisConfig {
          "_values": Object {
            "addLegend": true,
            "addTooltip": true,
            "alerts": Array [],
            "categoryAxes": Array [
              Object {
                "id": "CategoryAxis-1",
                "position": "top",
                "scale": Object {
                  "type": "MONTHS",
                },
                "type": "category",
              },
              Object {
                "id": "CategoryAxis-2",
                "position": "left",
                "scale": Object {
                  "type": "WEEKS",
                },
                "type": "category",
              },
            ],
            "colorSchema": "Greens",
            "colorsNumber": 4,
            "colorsRange": Array [],
            "enableHover": false,
            "grid": Object {
              "cellSize": 15,
              "style": "rect",
              "xOffset": 20,
              "yOffset": 20,
            },
            "handleNoResults": false,
            "invertColors": false,
            "legend": Object {
              "labels": Array [
                "0 - 4",
                "4 - 8",
                "8 - 12",
                "12 - 15",
              ],
            },
            "legendPosition": "right",
            "locale": "en_US",
            "percentageMode": false,
            "setColorRange": false,
            "style": Object {
              "margin": Object {
                "bottom": 5,
                "left": 3,
                "right": 3,
                "top": 10,
              },
            },
            "times": Array [],
            "type": "heatmap_year",
            "valueAxes": Array [
              Object {
                "id": "ValueAxis-1",
                "labels": Object {
                  "show": false,
                },
                "scale": Object {
                  "defaultYExtents": false,
                  "type": "linear",
                },
                "type": "value",
              },
            ],
          },
          "chartType": null,
          "vis": Object {
            "API": Object {
              "timeFilter": Object {
                "getBounds": [Function],
              },
            },
            "params": Object {
              "addLegend": true,
              "addTooltip": true,
              "categoryAxes": Array [
                Object {
                  "id": "CategoryAxis-1",
                  "position": "top",
                  "scale": Object {
                    "type": "MONTHS",
                  },
                  "type": "category",
                },
                Object {
                  "id": "CategoryAxis-2",
                  "position": "left",
                  "scale": Object {
                    "type": "WEEKS",
                  },
                  "type": "category",
                },
              ],
              "colorSchema": "Greens",
              "colorsNumber": 4,
              "colorsRange": Array [],
              "enableHover": false,
              "grid": Object {
                "cellSize": 15,
                "style": "rect",
                "xOffset": 20,
                "yOffset": 20,
              },
              "handleNoResults": false,
              "invertColors": false,
              "legendPosition": "right",
              "locale": "en_US",
              "percentageMode": false,
              "setColorRange": false,
              "times": Array [],
              "type": "heatmap_year",
              "valueAxes": Array [
                Object {
                  "id": "ValueAxis-1",
                  "labels": Object {
                    "show": false,
                  },
                  "scale": Object {
                    "defaultYExtents": false,
                    "type": "linear",
                  },
                  "type": "value",
                },
              ],
            },
          },
        }
      }
    >
      <div
        className="legend-col-wrapper"
      >
        <button
          aria-expanded={false}
          aria-label="Toggle legend"
          className="kuiCollapseButton legend-collapse-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="kuiIcon fa-chevron-circle-left"
          />
        </button>
      </div>
    </LegendBar>
  </div>
</div>
`;

exports[`LegendBar should render a full LegendBar component, default position to the right 1`] = `
<div
  className="vislib-container calendar-vis-container"
>
  <div
    className="visualize-legend calendar-vis-legend"
  >
    <LegendBar
      colorFunc={[Function]}
      dispatcher={
        Dispatcher {
          "API": null,
          "config": Object {
            "get": [Function],
          },
          "container": null,
          "dataTarget": "[data-label]",
          "src": null,
        }
      }
      getUiState={[Function]}
      position="vis-container--legend-right"
      renderComplete={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      setUiState={[Function]}
      visConfig={
        CalendarVisConfig {
          "_values": Object {
            "addLegend": true,
            "addTooltip": true,
            "alerts": Array [],
            "categoryAxes": Array [
              Object {
                "id": "CategoryAxis-1",
                "position": "top",
                "scale": Object {
                  "type": "MONTHS",
                },
                "type": "category",
              },
              Object {
                "id": "CategoryAxis-2",
                "position": "left",
                "scale": Object {
                  "type": "WEEKS",
                },
                "type": "category",
              },
            ],
            "colorSchema": "Greens",
            "colorsNumber": 4,
            "colorsRange": Array [],
            "enableHover": false,
            "grid": Object {
              "cellSize": 15,
              "style": "rect",
              "xOffset": 20,
              "yOffset": 20,
            },
            "handleNoResults": false,
            "invertColors": false,
            "legend": Object {
              "labels": Array [
                "0 - 4",
                "4 - 8",
                "8 - 12",
                "12 - 15",
              ],
            },
            "legendPosition": "right",
            "locale": "en_US",
            "percentageMode": false,
            "setColorRange": false,
            "style": Object {
              "margin": Object {
                "bottom": 5,
                "left": 3,
                "right": 3,
                "top": 10,
              },
            },
            "times": Array [],
            "type": "heatmap_year",
            "valueAxes": Array [
              Object {
                "id": "ValueAxis-1",
                "labels": Object {
                  "show": false,
                },
                "scale": Object {
                  "defaultYExtents": false,
                  "type": "linear",
                },
                "type": "value",
              },
            ],
          },
          "chartType": null,
          "vis": Object {
            "API": Object {
              "timeFilter": Object {
                "getBounds": [Function],
              },
            },
            "params": Object {
              "addLegend": true,
              "addTooltip": true,
              "categoryAxes": Array [
                Object {
                  "id": "CategoryAxis-1",
                  "position": "top",
                  "scale": Object {
                    "type": "MONTHS",
                  },
                  "type": "category",
                },
                Object {
                  "id": "CategoryAxis-2",
                  "position": "left",
                  "scale": Object {
                    "type": "WEEKS",
                  },
                  "type": "category",
                },
              ],
              "colorSchema": "Greens",
              "colorsNumber": 4,
              "colorsRange": Array [],
              "enableHover": false,
              "grid": Object {
                "cellSize": 15,
                "style": "rect",
                "xOffset": 20,
                "yOffset": 20,
              },
              "handleNoResults": false,
              "invertColors": false,
              "legendPosition": "right",
              "locale": "en_US",
              "percentageMode": false,
              "setColorRange": false,
              "times": Array [],
              "type": "heatmap_year",
              "valueAxes": Array [
                Object {
                  "id": "ValueAxis-1",
                  "labels": Object {
                    "show": false,
                  },
                  "scale": Object {
                    "defaultYExtents": false,
                    "type": "linear",
                  },
                  "type": "value",
                },
              ],
            },
          },
        }
      }
    >
      <div
        className="legend-col-wrapper"
      >
        <button
          aria-expanded={true}
          aria-label="Toggle legend"
          className="kuiCollapseButton legend-collapse-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="kuiIcon fa-chevron-circle-right"
          />
        </button>
        <ul
          className="legend-ul"
        >
          <li
            className="legend-value color"
            data-label="0 - 4"
            key="0"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <Legend
              colorFunc={[Function]}
              legendData={
                Object {
                  "label": "0 - 4",
                }
              }
            >
              <div
                className="legend-value-container"
              >
                <div
                  aria-label="0 - 4, toggle options"
                  className="legend-value-title legend-value-truncate"
                  kbn-accessible-click="true"
                >
                  <i
                    className="fa fa-circle"
                    style={
                      Object {
                        "color": "rgb(247,252,245)",
                      }
                    }
                  />
                  0 - 4
                </div>
              </div>
            </Legend>
          </li>
          <li
            className="legend-value color"
            data-label="4 - 8"
            key="1"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <Legend
              colorFunc={[Function]}
              legendData={
                Object {
                  "label": "4 - 8",
                }
              }
            >
              <div
                className="legend-value-container"
              >
                <div
                  aria-label="4 - 8, toggle options"
                  className="legend-value-title legend-value-truncate"
                  kbn-accessible-click="true"
                >
                  <i
                    className="fa fa-circle"
                    style={
                      Object {
                        "color": "rgb(199,233,192)",
                      }
                    }
                  />
                  4 - 8
                </div>
              </div>
            </Legend>
          </li>
          <li
            className="legend-value color"
            data-label="8 - 12"
            key="2"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <Legend
              colorFunc={[Function]}
              legendData={
                Object {
                  "label": "8 - 12",
                }
              }
            >
              <div
                className="legend-value-container"
              >
                <div
                  aria-label="8 - 12, toggle options"
                  className="legend-value-title legend-value-truncate"
                  kbn-accessible-click="true"
                >
                  <i
                    className="fa fa-circle"
                    style={
                      Object {
                        "color": "rgb(116,196,118)",
                      }
                    }
                  />
                  8 - 12
                </div>
              </div>
            </Legend>
          </li>
          <li
            className="legend-value color"
            data-label="12 - 15"
            key="3"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <Legend
              colorFunc={[Function]}
              legendData={
                Object {
                  "label": "12 - 15",
                }
              }
            >
              <div
                className="legend-value-container"
              >
                <div
                  aria-label="12 - 15, toggle options"
                  className="legend-value-title legend-value-truncate"
                  kbn-accessible-click="true"
                >
                  <i
                    className="fa fa-circle"
                    style={
                      Object {
                        "color": "rgb(35,139,69)",
                      }
                    }
                  />
                  12 - 15
                </div>
              </div>
            </Legend>
          </li>
        </ul>
      </div>
    </LegendBar>
  </div>
</div>
`;
